7.js

```diff
@@ -0,0 +1,61 @@
+const months = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun',
+'Jul', 'Aug', 'Sep', 'Oct', 'Nov',
+'Dec']
+
+function formatDate (date) {
+  if (date.includes('-')) {
+    return date
+  } else if (date.includes('/')) {
+    const dateArray = date.split('/')
+    return `${fullYear(dateArray[2])}-${dateArray[0]}-${dateArray[1]}`
+  } else if (date.includes('#')) {
+    const dateArray = date.split('#')
+    return `${fullYear(dateArray[1])}-${dateArray[0]}-${dateArray[2]}`
+  } else if (date.includes('*')) {
+    const dateArray = date.split('*')
+    return `${dateArray[2]}-${dateArray[1]}-${dateArray[0]}`
+  } else {
+    months.forEach(month => {
```

Comment:
This `else` block alone looks like it provides an opportunity for refactoring into a method.